### PR TITLE
drivers/mpsl/flash_synch: FTPAN-242 workaround

### PR DESCRIPTION
**Pull request by nvlsianpu**
- support in-progress abort of flash operation
- introduces https://github.com/nrfconnect/sdk-zephyr/pull/426

### CODE REVIEW

File: drivers/mpsl/flash_sync/flash_sync_mpsl.c [hunk 1]
```diff
@@ -80,9 +81,9 @@ timeslot_callback(mpsl_timeslot_session_id_t session_id, uint32_t signal)
 
 	switch (signal) {
 	case MPSL_TIMESLOT_SIGNAL_START:
-		if (_context.op_desc->handler(_context.op_desc->context) ==
-		    FLASH_OP_DONE) {
-			_context.status = 0;
+		rc = _context.op_desc->handler(_context.op_desc->context);
+		if (rc != FLASH_OP_ONGOING) {
+			_context.status = (rc == FLASH_OP_DONE) ? 0 : rc;
```
Comment (eugmes) [inline drivers/mpsl/flash_sync/flash_sync_mpsl.c:86]: `_context.status` should be something from `errno.h` negated. `FLASH_OP_ABORTED` will not do.

**Review by eugmes — CHANGES_REQUESTED** [review]

File: drivers/mpsl/flash_sync/flash_sync_mpsl.c (hunk 1 above)
Comment (nvlsianpu) [reply drivers/mpsl/flash_sync/flash_sync_mpsl.c:86]: fixed in the handler() patch

**Comment by nvlsianpu** [comment]
@eugmes Please revisit

**Review by eugmes — APPROVED** [review]
Looks ok now.

**Review by tejlmand — CHANGES_REQUESTED** [review]
Included in: #3895

**Comment by nvlsianpu** [comment]
Already Merged via #3895

**Pull request closed without merging**